src/extension/content-script/batteries/GitHub.ts

```diff
@@ -3,75 +3,115 @@ import setLightningData from "../setLightningData";
 
 const urlMatcher = /^https:\/\/github.com\/([^/]+)(\/([^/]+))?$/;
 
-const battery = (): void => {
+const battery = async (): Promise<void> => {
   const urlParts = document.location.pathname.split("/");
   const username = urlParts[1];
   const repo = urlParts[2];
 
+  let found = false;
+
   if (username && repo) {
-    handleRepositoryPage(username);
-  } else if (username) {
-    handleProfilePage();
+    found = await handleRepositoryPage(username, repo);
   }
-};
 
-function parseElement(elementSelector: string) {
-  const text = document.querySelector<HTMLElement>(elementSelector)?.innerText;
-  if (text) {
-    const match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@\S+)/i);
-    if (match) return match[2];
+  if (!found && username) {
+    await handleProfilePage(username);
   }
+};
+
+function parseText(text: string) {
```

Comment:
is this i.e. `findLightningAddressInText`?

Reply:
Yes it was called parseElement before. 
I've renamed it to findLightningAddressInText as you suggested

Reply:
Ah ok, missed that, sorry. Thanks for changing it!